Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,148 @@
+        
+        _dynamicEntityLayer = .init(initialValue: DynamicEntityLayer(dataSource: customDataSource))
+        
+        // Set display tracking properties on the layer.
+        let trackDisplayProperties = dynamicEntityLayer.trackDisplayProperties
+        trackDisplayProperties.showsPreviousObservations = true
+        trackDisplayProperties.showsTrackLine = true
+        trackDisplayProperties.maximumObservations = 20
+        
+        // Create the label definition so we can show the vessel name on top of
+        // each dynamic entity.
+        let labelDefinition = LabelDefinition(
+            labelExpression: SimpleLabelExpression(simpleExpression: "[VesselName]"),
+            textSymbol: TextSymbol(color: .red, size: 12)
+        )
+        labelDefinition.placement = .pointAboveCenter
+        
+        dynamicEntityLayer.addLabelDefinition(labelDefinition)
+        dynamicEntityLayer.labelsAreEnabled = true
+        
+        map.addOperationalLayer(dynamicEntityLayer)
+    }
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: map)
+                .onSingleTapGesture { screenPoint, _ in
+                    tappedScreenPoint = screenPoint
+                }
+                .callout(placement: $calloutPlacement.animation(.default.speed(2))) { placement in
+                    let attributes = (placement.geoElement as! DynamicEntityObservation).attributes
+                    VStack(alignment: .leading) {
+                        // Display all the attributes in the callout.
+                        ForEach(attributes.sorted(by: { $0.key < $1.key }), id: \.key) { item in
+                            Text("\(item.key): \(String(describing: item.value))")
+                        }
+                    }
+                }
+                .task(id: tappedScreenPoint) {
+                    let newCalloutPlacement: CalloutPlacement?
+                    if let tappedScreenPoint,
+                       let identifyResult = try? await proxy.identify(
+                        on: dynamicEntityLayer,
+                        screenPoint: tappedScreenPoint,
+                        tolerance: 2
+                       ) {
+                        // Set the callout placement to the observation that was tapped on.
+                        newCalloutPlacement = identifyResult.geoElements.first.map { .geoElement($0) }
+                    } else {
+                        // Hides the callout.
+                        newCalloutPlacement = nil
+                    }
+                    calloutPlacement = newCalloutPlacement
+                }
+        }
+    }
+}
+
+/// The vessel feed that is emitting custom dynamic entity events.
+private struct VesselFeed: CustomDynamicEntityFeed {
+    typealias Events = AsyncThrowingMapSequence<AsyncLineSequence<URL.AsyncBytes>, CustomDynamicEntityFeedEvent>
+    
+    let events = URL.selectedVesselsDataSource.lines.map { line in
```

Comment:
`Events` isn't needed as it is inferred by the compiler from the type of `events`.

```suggestion
private struct VesselFeed: CustomDynamicEntityFeed {
    let events = URL.selectedVesselsDataSource.lines.map { line in
```

Reply:
Yeah, I tried that and I was getting a compile error saying `VesselFeed` did not conform to `CustomDynamicEntityFeed`.

Reply:
Which version of Xcode are you using? It works fine in Xcode 14.1.

Reply:
Welp, just kidding I tried again and the compile error went away.

Reply:
I was using 14.3.1 and the compile error would not go away but it did now so all good.